### DISCUSSION TITLE
Access for spring-boot-data-aerospike

### DISCUSSION THREAD

**Discussion by Aloren** [aerospike/spring-data-aerospike#534] [Team Posts]
Please provide Admin access for Aloren user to repo: https://github.com/aerospike-community/spring-boot-data-aerospike as for spring-data-aerospike. This repo contains autoconfiguration module of spring-data-aerospike.
Thank you.

**Comment by rbotzer** [comment]
Done. Is this going to be the Spring Boot Starter for using Aerospike with Spring Data?

**Comment by tdanylchuk** [comment]
yes, kind of, so user don't need to create aerospike client manually, but spring autoconfiguration will do

**Comment by rbotzer** [comment]
Got it. Did you consider working on an actual Spring Boot Starter? We've heard from a couple of people that it's an important missing piece for bootstrapping new projects. I was wondering what your thoughts were on the topic.

**Comment by tdanylchuk** [comment]
what do you mean under "Spring Boot Starter"?

**Comment by rbotzer** [comment]
We got feedback from a CTO of a large training company that you can get a Spring Boot application with a Mongo backend scaffolded in minutes, but that it doesn't work with Aerospike. He pointed out there is no `spring-boot-starter-data-aerospike` similar to `spring-boot-starter-data-mongodb`. 

Similarly, this thread on our discussion forum has been open for a while, but may be outdated: https://discuss.aerospike.com/t/how-to-use-aerospike-in-spring-framework-like-mongodb/2579

I'm not sure yet if the issue is missing components, lack of compatibility with newer versions of Spring Boot, or what. Any input from you is useful, but eventually I'll ask a developer here to look into it and assess it. Do you know what he was referring to?  He claimed to have dug into it, read the blog posts on Medium, and still unable to roll out an application with an Aerospike backend as easily as with Mongo.

**Comment by tdanylchuk** [comment]
this module will take handle it aslike in Mongo (but it will require support, since some new parameters could be added to configure Aerospike client)

**Comment by tdanylchuk** [comment]
ideally you wont need to add anything related to aerospike to branch new project except 
```
<dependency>
  <groupId>com.aerospike</groupId>
  <artifactId>spring-boot-starter-data-aerospike</artifactId>  
</dependency>
```

and configuration
```
spring.data.aerospike.hosts=127.0.0.1:3000
spring.data.aerospike.namespace=COMMON
```

then your application internally won't even notice it uses aerospike under the hood, meaning users could seamlessly switch apps from Mongo to Aerospike

**Comment by rbotzer** [comment]
Thanks, that helps.

**Comment by Aloren** [comment]
@rbotzer This repo provides starter for Aerospike: https://github.com/aerospike-community/spring-boot-data-aerospike/tree/master/spring-boot-starter-data-aerospike 
I will update my demo project and create a new article for Medium, so that the link to it can be posted on the forum.
